cdap-common/src/main/java/co/cask/cdap/common/namespace/RemoteNamespaceQueryClient.java

```diff
@@ -42,26 +48,41 @@
  */
 public class RemoteNamespaceQueryClient extends AbstractNamespaceQueryClient {
   private final Supplier<EndpointStrategy> endpointStrategySupplier;
+  private final String userId;
 
   @Inject
-  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient) {
+  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient, CConfiguration cConf) {
     this.endpointStrategySupplier = Suppliers.memoize(new Supplier<EndpointStrategy>() {
       @Override
       public EndpointStrategy get() {
-        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.REMOTE_SYSTEM_OPERATION));
+        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.APP_FABRIC_HTTP));
       }
     });
+    try {
+      if (SecurityUtil.isKerberosEnabled(cConf)) {
+        // if kerberos is enabled use the name from cdap master principal
+        String masterPrincipal = cConf.get(Constants.Security.CFG_CDAP_MASTER_KRB_PRINCIPAL);
```

Comment:
What is this for? This is a remote client, which means it should haven't Kerberos login. Should it just like any other http client that set the USER_ID in the header?